huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go

```diff
@@ -441,18 +520,58 @@ func resourceDmsKafkaInstanceUpdate(ctx context.Context, d *schema.ResourceData,
 		engine := "kafka"
 		tagErr := utils.UpdateResourceTags(dmsV2Client, d, engine, d.Id())
 		if tagErr != nil {
-			return fmtp.DiagErrorf("error updating tags of dms kafka instance:%s, err:%s", d.Id(), tagErr)
+			e := fmtp.Errorf("error updating tags of DMS kafka instance:%s, err:%s", d.Id(), tagErr)
+			mErr = multierror.Append(mErr, e)
 		}
 	}
-
+	if mErr.ErrorOrNil() != nil {
+		return fmtp.DiagErrorf("error while updating DMS Kafka instances, there %s", mErr)
+	}
 	return resourceDmsKafkaInstanceRead(ctx, d, meta)
 }
 
+func resizeInstance(ctx context.Context, d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	dmsV2Client, err := config.DmsV2Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.Errorf("error creating HuaweiCloud DMS instance client: %s", err)
+	}
+
+	product, err := getProductDetail(config, d)
+	if err != nil || product == nil {
+		return fmtp.Errorf("change storage_space failed, error querying product detail: %s", err)
```

Comment:
If the error return of function `getProfuctDetail` is null and the product is empty, the error is reported as `change storage_space failed, error querying product detail: <nil>`, which is confusing

Reply:
see `getProfuctDetail`, this case will not happen

Reply:
does the `product == nil` needed?